velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -142,14 +142,16 @@ std::shared_ptr<WriterProperties> getArrowParquetWriterOptions(
 Writer::Writer(
     std::unique_ptr<dwio::common::FileSink> sink,
     const WriterOptions& options,
-    std::shared_ptr<memory::MemoryPool> pool)
+    std::shared_ptr<memory::MemoryPool> pool,
+    RowTypePtr schema)
     : pool_(std::move(pool)),
       generalPool_{pool_->addLeafChild(".general")},
       stream_(std::make_shared<ArrowDataBufferSink>(
           std::move(sink),
           *generalPool_,
           options.bufferGrowRatio)),
-      arrowContext_(std::make_shared<ArrowContext>()) {
+      arrowContext_(std::make_shared<ArrowContext>()),
+      schema_(std::move(schema)) {
```

Comment:
Add a check that schema has non-empty unique field names.

ROW({BIGINT(), VARCHAR()} and ROW({"a", "a"} ,{BIGINT(), VARCHAR()} should fail.